From 3fe471d34ef6575bc49c9f85ed74368e3ac25ad6 Mon Sep 17 00:00:00 2001
From: Jason Kridner <jdk@ti.com>
Date: Sun, 18 Sep 2011 12:16:31 -0400
Subject: [PATCH 15/21] HACK: beagleboard: config: disable cache for USB

There is currently a bug in the USB code for the BeagleBoard that is
worked-around by disabling the cache.

See http://comments.gmane.org/gmane.comp.boot-loaders.u-boot/108224

Signed-off-by: Jason Kridner <jdk@ti.com>
---
 include/configs/omap3_beagle.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
index 941ec38..569d747 100644
--- a/include/configs/omap3_beagle.h
+++ b/include/configs/omap3_beagle.h
@@ -36,6 +36,10 @@
 #define CONFIG_OMAP3430		1	/* which is in a 3430 */
 #define CONFIG_OMAP3_BEAGLE	1	/* working with BEAGLE */
 
+/* TODO: WORK-AROUND for bug in BeagleBoard-specific USB */
+#define CONFIG_SYS_ICACHE_OFF
+#define CONFIG_SYS_DCACHE_OFF
+
 #define CONFIG_SDRC	/* The chip has SDRC controller */
 
 #include <asm/arch/cpu.h>		/* get chip and board defs */
-- 
1.7.2.5

